sdcm/nemesis.py

```diff
@@ -482,6 +483,42 @@ def disrupt_rolling_restart_cluster(self, random=False):
         self._set_current_disruption('RollingRestartCluster %s' % self.target_node)
         self.cluster.restart_scylla(random=random)
 
+    def disrupt_switch_between_PasswordAuthenticator_and_SaslauthdAuthenticator_and_back(self):
+        """
+        If prepare_saslauthd is enabled, saslauthd and ldap environment will be prepared for
+        using SaslauthdAuthenticator. We have same account (cassandra) for SaslauthdAuthenticator
+        and PasswordAuthenticator, so it can be toggled smoothly without effect to c-s workloads.
+
+        It's only support to switch between PasswordAuthenticator and SaslauthdAuthenticator,
+        the authenticator will be reset back in the end of nemesis.
+        """
+        self._set_current_disruption('SwitchBetweenPasswordAuthAndSaslauthdAuth %s' % self.target_node)
+        result = self.target_node.remoter.run("grep -o '^authenticator: .*' /etc/scylla/scylla.yaml")
+        if 'com.scylladb.auth.SaslauthdAuthenticator' in result.stdout:
+            orig_auth = 'com.scylladb.auth.SaslauthdAuthenticator'
+            opposite_auth = 'PasswordAuthenticator'
+        elif 'PasswordAuthenticator' in result.stdout:
+            orig_auth = 'PasswordAuthenticator'
+            opposite_auth = 'com.scylladb.auth.SaslauthdAuthenticator'
+        else:
+            raise UnsupportedNemesis(
+                'This nemesis only supports to switch between SaslauthdAuthenticator and PasswordAuthenticator')
+        if self.cluster.params.get('prepare_saslauthd'):
+            update_authenticator(self.cluster.nodes, opposite_auth)
+            try:
+                # Run connect a new session after authenticator switch, and run a short workload
+                self._prepare_test_table(ks='keyspace_for_authenticator_switch', table='standard1')
+            finally:
+                time.sleep(120)
+                update_authenticator(self.cluster.nodes, orig_auth)
+
+            # Run connect a new session after authenticator switch, drop the test keyspace
+            with self.cluster.cql_connection_patient(self.target_node) as session:
+                session.execute(f'DROP KEYSPACE keyspace_for_authenticator_switch')
```

Comment:
shouldn't this be in the `finally` statement, as for either success or exception, we drop this KS used for this test only?

Reply:
You are right, fixed.